### PR TITLE
Document sun textures/displaylist

### PR DESCRIPTION
**Pull request by tom-overton**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

There was some talk about how the sun worked in OoT today, so I ported over the OoT names into MM. According to Emil, these should actually be a single 64x64 texture, but that doesn't quite work with ZAPD, so I left them as three separate textures for now.

### CODE REVIEW

File: assets/xml/objects/gameplay_keep.xml [hunk 1]
```diff
@@ -1316,15 +1316,15 @@
...
-        <Texture Name="gameplay_keep_Tex_07A0F0" OutName="tex_07A0F0" Format="i4" Width="64" Height="17" Offset="0x7A0F0" />
-        <Texture Name="gameplay_keep_Tex_07A310" OutName="tex_07A310" Format="i4" Width="64" Height="31" Offset="0x7A310" />
-        <Texture Name="gameplay_keep_Tex_07A6F0" OutName="tex_07A6F0" Format="i4" Width="64" Height="16" Offset="0x7A6F0" />
-        <Texture Name="gameplay_keep_Tex_07A8F0" OutName="tex_07A8F0" Format="i4" Width="64" Height="17" Offset="0x7A8F0" />
+        <Texture Name="gSun1Tex" OutName="sun_1" Format="i4" Width="64" Height="31" Offset="0x79B10" />
+        <Texture Name="gSun2Tex" OutName="sun_2" Format="i4" Width="64" Height="16" Offset="0x79EF0" />
+        <Texture Name="gSun3Tex" OutName="sun_3" Format="i4" Width="64" Height="17" Offset="0x7A0F0" />
+        <Texture Name="gSunEvening1Tex" OutName="sun_evening_1" Format="i4" Width="64" Height="31" Offset="0x7A310" />
+        <Texture Name="gSunEvening2Tex" OutName="sun_evening_2" Format="i4" Width="64" Height="16" Offset="0x7A6F0" />
+        <Texture Name="gSunEvening3Tex" OutName="sun_evening_3" Format="i4" Width="64" Height="17" Offset="0x7A8F0" />
```
Comment (engineer124) [inline assets/xml/objects/gameplay_keep.xml:1324]: OoT has `gSunEvening3Tex` has a `Height="16"`, is that difference intentional?

`"gSunEvening3Tex" OutName="sun_evening_3" Format="i4" Width="64" Height="16"`
https://github.com/zeldaret/oot/blob/eadc477187888e1ae078d021b4a00b1366f0c9a4/assets/xml/objects/gameplay_keep.xml#L933

Comment (tom-overton) [reply assets/xml/objects/gameplay_keep.xml:1324]: According to Emil:
> There are also 32 bits "padding" after the end of the last texture part, which should really be the last row in the 64x64 texture.

So I think the only real difference between the way OoT is doing it and the way MM is doing it is that MM's version will put the padding in the texture itself, whereas OoT's will put it in the .c file as a bunch of unaccounted-for 0x00s.

**Review by engineer124 — APPROVED** [review]

File: assets/xml/objects/gameplay_keep.xml (hunk 1 above)
Comment (tom-overton) [reply assets/xml/objects/gameplay_keep.xml:1324]: Let me put a comment here explaining that these *should* be 64x64, but ZAPD currently doesn't handle it. That should hopefully explain the weird 17-pixel tall texture.

**Review by jpburnett — APPROVED** [review]
\T/

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**